File: CanaryDemo/ObjectiveCDemo/CanaryTTYLogger.m
//
//  CanaryTTYLogger.m
//  ObjectiveCDemo
//
//  Created by Rake Yang on 2020/12/13.
//

#import "CanaryTTYLogger.h"
#import <Canary/Canary-Swift.h>

@implementation CanaryTTYLogger

+ (instancetype)shared {
    static dispatch_once_t onceToken;
    static CanaryTTYLogger *_instance;
    dispatch_once(&onceToken, ^{
        _instance = [[self alloc] init];
    });
    return _instance;
}

- (void)logMessage:(DDLogMessage *)logMessage {
    [CanarySwift.shared storeLogMessageWithDict:[logMessage dictionaryWithValuesForKeys:@{}] timestamp:logMessage.timestamp.timeIntervalSince1970];
}

@end

File: Examples/iOSDemo/CanaryDemo/CanaryTTYLogger.swift
//
//  CanaryTTYLogger.swift
//  CanaryDemo
//
//  Created by Rake Yang on 2020/12/25.
//

import Foundation
import Canary
import CocoaLumberjack

public class CanaryTTYLogger: DDAbstractLogger {
    static let shared = CanaryTTYLogger()
    public override func log(message logMessage: DDLogMessage) {
        CanaryManager.shared.storeLogMessage(dict: logMessage.dictionaryWithValues(forKeys: CanaryManager.StoreLogKeys), timestamp: logMessage.timestamp.timeIntervalSince1970)
    }
}
